cpp/src/parquet/chunker_internal_test.cc

```diff
@@ -0,0 +1,1397 @@
+
+  // Verify elements before and after the difference are identical
+  for (size_t i = 0; i < 4; i++) {
+    ASSERT_EQ(first[i], second[i]);
+  }
+  for (size_t i = 9; i < first.size(); i++) {
+    ASSERT_EQ(first[i], second[i]);
+  }
+}
+
+TEST(TestFindDifferences, AdditionalCase) {
+  ChunkList original = {445, 312, 393, 401, 410, 138, 558, 457};
+  ChunkList modified = {445, 312, 393, 393, 410, 138, 558, 457};
+
+  auto diffs = FindDifferences(original, modified);
+  ASSERT_EQ(diffs.size(), 1);
+
+  ASSERT_EQ(diffs[0].first, ChunkList({401}));
+  ASSERT_EQ(diffs[0].second, ChunkList({393}));
+
+  // Verify elements before and after the difference are identical
+  for (size_t i = 0; i < 3; i++) {
+    ASSERT_EQ(original[i], modified[i]);
+  }
+  for (size_t i = 4; i < original.size(); i++) {
+    ASSERT_EQ(original[i], modified[i]);
+  }
+}
+
+void AssertPageLengthDifferences(const RowGroupInfo& original,
+                                 const RowGroupInfo& modified,
+                                 int8_t exact_number_of_equal_diffs,
+                                 int8_t exact_number_of_larger_diffs,
+                                 int8_t exact_number_of_smaller_diffs,
+                                 int64_t edit_length = 0) {
+  // Asserts that the differences between the original and modified page lengths
+  // are as expected. A longest common subsequence diff is calculated on the original
+  // and modified sequences of page lengths. The exact_number_of_equal_diffs,
+  // exact_number_of_larger_diffs, and exact_number_of_smaller_diffs parameters specify
+  // the expected number of differences with equal, larger, and smaller sums of the page
+  // lengths respectively. The edit_length parameter is used to verify that the page
+  // lenght differences are exactly equal to the edit_length.
+  auto diffs = FindDifferences(original.page_lengths, modified.page_lengths);
+  size_t expected_number_of_diffs = exact_number_of_equal_diffs +
+                                    exact_number_of_larger_diffs +
+                                    exact_number_of_smaller_diffs;
+  if (diffs.size() != expected_number_of_diffs) {
+    PrintDifferences(original.page_lengths, modified.page_lengths, diffs);
+  }
+  if (diffs.size() == 0) {
+    // no differences found, the arrays are equal
+    ASSERT_TRUE(original.page_lengths == modified.page_lengths);
+  }
+  ASSERT_EQ(diffs.size(), expected_number_of_diffs);
+
+  uint8_t equal_diffs = 0;
+  int8_t larger_diffs = 0;
+  int8_t smaller_diffs = 0;
+  for (const auto& diff : diffs) {
+    uint64_t original_sum = 0, modified_sum = 0;
+    for (const auto& val : diff.first) original_sum += val;
+    for (const auto& val : diff.second) modified_sum += val;
+
+    if (original_sum == modified_sum) {
+      equal_diffs++;
+    } else if (original_sum < modified_sum) {
+      larger_diffs++;
+      ASSERT_EQ(original_sum + edit_length, modified_sum);
```

Comment:
I don't understand: all edits are supposed to have the same edit length?

Reply:
Yes, this is an assumption by the assertion function, I added a comment about this with a note to improve it in the future.